pkg/controllerhelpers/handlers.go

```diff
@@ -89,29 +89,41 @@ func (h *Handlers[T]) EnqueueAll(depth int, untypedObj kubeinterfaces.ObjectInte
 	}
 }
 
-func (h *Handlers[QT]) EnqueueOwner(depth int, obj kubeinterfaces.ObjectInterface, operation HandlerOperationType) {
-	controllerRef := metav1.GetControllerOf(obj)
-	if controllerRef == nil {
-		return
-	}
+func (h *Handlers[QT]) EnqueueOwnerIf(precondition func(QT) bool) EnqueueFuncType {
+	return func(depth int, obj kubeinterfaces.ObjectInterface, operation HandlerOperationType) {
+		controllerRef := metav1.GetControllerOf(obj)
+		if controllerRef == nil {
+			return
+		}
 
-	if controllerRef.Kind != h.gvk.Kind {
-		return
-	}
+		if controllerRef.Kind != h.gvk.Kind {
+			return
+		}
 
-	owner, err := h.getterLister.Get(obj.GetNamespace(), controllerRef.Name)
-	if err != nil {
-		utilruntime.HandleError(err)
-		return
+		owner, err := h.getterLister.Get(obj.GetNamespace(), controllerRef.Name)
+		if err != nil {
+			utilruntime.HandleError(err)
+			return
+		}
+
+		if owner.GetUID() != controllerRef.UID {
+			utilruntime.HandleError(err)
+			return
+		}
+
+		if precondition(owner) {
+			klog.V(4).InfoSDepth(depth, "Enqueuing owner", append([]any{"OwnerGVK", h.gvk, "OwnerRef", klog.KObj(owner), "OwnerUID", owner.GetUID()}, getObjectLogContext(obj, nil)...)...)
```

Comment:
nit: would it be better to make the function extendable and use  `'!precondition` pattern?